gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/BucketRegion.java

```diff
@@ -910,6 +910,7 @@ private void destroyAllEntries(Set keysToDestroy) {
       event.setBatchUUID(this.batchUUID); // to make sure that lock is not nexessary
 
       if (getTXState() != null) {
+        event.setRegion(this);
         getTXState().destroyExistingEntry(event, true, null);
```

Comment:
getTXState() is a relatively expensive thread-local map lookup. Never invoke this method multiple times in a code block. Same holds for all thread local lookups (e.g. snapshot TX lookups).

The event is supposed to carry the TXState so that other layers don't have to again do thread-local lookups. Use something like:

```java
TXStateInterface txState = event.getTXState(this);
if (txState != null) {
  event.setRegion(this);
  txState.destroyExistingEntry(...);
} else {
...
```